File: src/main/java/lv/ldn/scalaoop/java/Deserializer.java
package lv.ldn.scalaoop.java;

import java.util.Date;

public class Deserializer {

    public <T> T deserialize(String data, Class<T> klass) {
        try {
            T instance = klass.newInstance();
            // deserialization magic
            return instance;
        } catch (InstantiationException | IllegalAccessException e) {
            throw new RuntimeException(e);
        }
    }

    public static void main(String[] args) {
        new Deserializer().deserialize("2014-02-15", Date.class);
    }

}

File: src/main/scala/lv/ldn/scalaoop/scala/Deserializer.scala
package lv.ldn.scalaoop.scala

import scala.reflect.ClassTag


class Deserializer {
  def deserialize[T](data: String)(implicit klass: ClassTag[T]): T = {
    val instance = klass.runtimeClass.newInstance().asInstanceOf[T]
    // deserialization magic
    instance
  }
}

object Deserializer {
  def main(args: Array[String]) {
    val date = new Deserializer().deserialize[java.util.Date]("2014-02-15")
  }
}
